src/Classifiers/OneVsRest.php

```diff
@@ -0,0 +1,254 @@
+     * @return bool
+     */
+    public function trained() : bool
+    {
+        return !empty($this->classifiers);
+    }
+
+    /**
+     * Train the learner with a dataset.
+     *
+     * @param \Rubix\ML\Datasets\Labeled $dataset
+     */
+    public function train(Dataset $dataset) : void
+    {
+        SpecificationChain::with([
+            new DatasetIsLabeled($dataset),
+            new DatasetIsNotEmpty($dataset),
+            new SamplesAreCompatibleWithEstimator($dataset, $this),
+            new LabelsAreCompatibleWithLearner($dataset, $this),
+        ])->check();
+
+        $classes = $dataset->possibleOutcomes();
+
+        $this->backend->flush();
+
+        foreach ($classes as $class) {
+            $estimator = clone $this->base;
+            $subset = clone $dataset;
+
+            $binarize = function ($label) use ($class) {
+                return $label === $class ? 'y' : 'n';
+            };
+
+            $subset->transformLabels($binarize);
+
+            $task = new TrainLearner($estimator, $subset);
+
+            $this->backend->enqueue($task);
+        }
+
+        $classifiers = $this->backend->process();
+
+        $classifiers = array_combine($classes, $classifiers) ?: [];
+
+        $this->classifiers = $classifiers;
+
+        $this->featureCount = $dataset->numFeatures();
+    }
+
+    /**
+     * Make predictions from a dataset.
+     *
+     * @param \Rubix\ML\Datasets\Dataset $dataset
+     * @throws \Rubix\ML\Exceptions\RuntimeException
+     * @return list<string>
+     */
+    public function predict(Dataset $dataset) : array
+    {
+        return array_map('\Rubix\ML\argmax', $this->proba($dataset));
+    }
+
+    /**
+     * Estimate the joint probabilities for each possible outcome.
+     *
+     * @param \Rubix\ML\Datasets\Dataset $dataset
+     * @throws \Rubix\ML\Exceptions\RuntimeException
+     * @return list<array<string,float>>
+     */
+    public function proba(Dataset $dataset) : array
+    {
+        if (!$this->classifiers or !$this->featureCount) {
+            throw new RuntimeException('Estimator has not been trained.');
+        }
+
+        DatasetHasDimensionality::with($dataset, $this->featureCount)->check();
+
+        $this->backend->flush();
+
+        /** @var \Rubix\ML\Probabilistic $estimator */
+        foreach ($this->classifiers as $estimator) {
+            $task = new Proba($estimator, $dataset);
+            
+            $this->backend->enqueue($task);
+        }
+
+        $aggregate = $this->backend->process();
+
+        $aggregate = array_transpose($aggregate);
+
+        $classes = array_keys($this->classifiers);
+
+        $probabilities = [];
+
+        foreach ($aggregate as $votes) {
+            $dist = [];
+
+            foreach ($votes as $j => $proba) {
+                $dist[$classes[$j]] = $proba['y'];
+            }
+
+            $total = array_sum($dist);
+
+            foreach ($dist as &$probability) {
+                $probability /= $total;
```

Comment:
What you are doing here looks like it should be a SoftMax. Did this formula come from some paper. My gut feeling is averaging the probabilities shouldn't really work. If you want to have this here it might work if you change line 228 to

```
$dist[$classes[$j]] = $proba['y'] > $proba['n'] ? 1.0 : 0.0;
```

But using a SoftMax will work better if you want to use `$proba[y]` in line 228.